### PR TITLE
Client-side qiskit job retrieval

### LINKED ISSUE #579: Get `qiskit-superstaq` job from `job_id`
### What is happening?

`qiskit-superstaq` doesn't offer a way to retrieve job from job ID. (i.e., like `service.get_job` in `cirq-superstaq`).

### What are the current alternatives?

None -- we should support `backend.retrieve_job` for `qiskit` parity as well as for `cirq-superstaq` parity.

### Any additional context?

_No response_

### PR DESCRIPTION
**Pull request by dowusu-antwi**
Fixes #579.

### CODE REVIEW

**Review by vtomole — APPROVED** [review]
LGTM

File: qiskit-superstaq/qiskit_superstaq/superstaq_backend.py [hunk 1]
```diff
@@ -120,6 +120,22 @@ def run(
...
 
+    def retrieve_job(self, job_id: str) -> qss.SuperstaqJob:
+        """Gets a job that has been created on the Superstaq API.
+
+        Args:
+            job_id: The UUID of the job. Jobs are assigned these numbers by the server during the
+            creation of the job.
+
+        Returns:
+            A `qss.Job` which can be queried for status or results.
```
Comment (bharat-thotakura) [inline qiskit-superstaq/qiskit_superstaq/superstaq_backend.py:131]: Should this be: 
```suggestion
            A `qss.SuperstaqJob` which can be queried for status or results.
```
?

**Review by bharat-thotakura — APPROVED** [review]
LGTM (just had one question)

**Comment by vtomole** [comment]
@dhirpalshah  will deploy once https://github.com/Infleqtion/client-superstaq/pull/558 is merged in.

**Pull request merged**